packages/vats/src/provisionPool.js

```diff
@@ -110,121 +51,25 @@ export const start = async (zcf, privateArgs) => {
-    const invitation = E(psmPub).makeWantMintedInvitation();
-    const seat = E(zoe).offer(invitation, proposal, { In: payIn });
-    const payout = await E(seat).getPayout('Out');
-    const rxd = await E(fundPurse).deposit(payout);
-    metrics.onTrade(rxd);
-    return rxd;
-  };
-
-  /** @type {MapStore<Brand, Instance>} */
-  const brandToPSM = makeScalarMapStore();
-
-  void observeIteration(E(poolBank).getAssetSubscription(), {
-    updateState: async desc => {
-      console.log('provisionPool notified of new asset', desc.brand);
-      await zcf.saveIssuer(desc.issuer, desc.issuerName);
-      /** @type {ERef<Purse>} */
-      // @ts-expect-error vbank purse is close enough for our use.
-      const exchangePurse = E(poolBank).getPurse(desc.brand);
-      void observeNotifier(E(exchangePurse).getCurrentAmountNotifier(), {
-        updateState: async amount => {
-          console.log('provisionPool balance update', amount);
-          if (AmountMath.isEmpty(amount) || amount.brand === poolBrand) {
-            return;
-          }
-          if (!brandToPSM.has(desc.brand)) {
-            console.error('funds arrived but no PSM instance', desc.brand);
-            return;
-          }
-          const instance = brandToPSM.get(desc.brand);
-          const payment = E(exchangePurse).withdraw(amount);
-          await swap(payment, amount, instance).catch(async reason => {
-            console.error(X`swap failed: ${reason}`);
-            const resolvedPayment = await payment;
-            return E(exchangePurse).deposit(resolvedPayment);
-          });
-        },
-        fail: reason => console.error(reason),
-      });
-    },
+  const makeProvisionPoolKit = prepareProvisionPoolKit(baggage, {
+    makeRecorderKit,
+    params,
+    poolBank,
+    zcf,
   });
 
-  /**
-   *
-   * @param {ERef<Bank>} destBank
-   */
-  const sendInitialPayment = async destBank => {
-    const perAccountInitialAmount = /** @type {Amount<'nat'>} */ (
-      params.getPerAccountInitialAmount()
-    );
-    const initialPmt = await E(fundPurse).withdraw(perAccountInitialAmount);
-
-    const destPurse = E(destBank).getPurse(poolBrand);
-    return E(destPurse)
-      .deposit(initialPmt)
-      .then(amt => {
-        metrics.onSendFunds(perAccountInitialAmount);
-        console.log('provisionPool sent', amt, 'to', address);
-      })
-      .catch(reason => {
-        console.error(
-          X`initial deposit for ${q(address)} failed: ${q(reason)}`,
-        );
-        void E(fundPurse).deposit(initialPmt);
-        throw reason;
-      });
-  };
+  const provisionPoolKit = await provideSingleton(
+    baggage,
+    'provisionPoolKit',
+    () =>
+      makeProvisionPoolKit({
+        // XXX governance can change the brand of the amount but should only be able to change the value
+        // NB: changing the brand will break this pool
+        poolBrand: params.getPerAccountInitialAmount().brand,
```

Comment:
Good to be clear about that, I suppose. The only useful brand is IST in any case.